enum.c

```diff
@@ -1069,10 +1069,13 @@ static VALUE
 enum_tally(int argc, VALUE *argv, VALUE obj)
 {
     VALUE hash;
-    if (rb_check_arity(argc, 0, 1))
-        hash = rb_check_hash_type(argv[0]);
-    else
+    if (rb_check_arity(argc, 0, 1)) {
+        hash = rb_convert_type(argv[0], T_HASH, "Hash", "to_hash");
+        rb_check_frozen(hash);
+    } else {
```

Comment:
```suggestion
    }
    else {
```

Reply:
🙇 Applied in https://github.com/ruby/ruby/pull/4327/commits/0b0814c912a144ee487cf6e81f141b601f2f5aa8!